src/components/NcAvatar/NcAvatar.vue

```diff
@@ -420,6 +420,16 @@ export default {
 			default: 'body',
 		},
 	},
+
+	setup() {
+		const root = useTemplateRef('main')
+		const isDarkTheme = useIsDarkThemeElement(root)
```

Comment:
I am not sure it is a performance improvement if every single avatar now adds mutation observer to support live theme change...

Reply:
Avatars are actively used in lists, mentions and etc.
This component should be lightweight.